Autopilot/Common/Interfaces/UART.c

```diff
@@ -123,77 +128,95 @@ void initUART(unsigned char interface, unsigned long int baudrate)
 
 void quoueTXData(unsigned char interface, unsigned char* data, unsigned int data_length)
 {
     unsigned int i;
-    if (interface == 1 && uart1_status == INTERFACE_INITIALIZED) {
+    if (interface == 1 && (uart1_status & UART_TX_ENABLE)) {
         for (i = 0; i < data_length; i++) {
             pushBQueue(&uart1_tx_queue, data[i]);
         }
-    } else if (interface == 2 && uart2_status == INTERFACE_INITIALIZED) {
+
+        if (U1STAbits.TRMT) { //if the transmit buffer is empty and register (no sending)
+            U1TXREG = popBQueue(&uart1_tx_queue); //trigger a send
+        }
+    } else if (interface == 2 && (uart2_status & UART_TX_ENABLE)) {
         for (i = 0; i < data_length; i++) {
             pushBQueue(&uart2_tx_queue, data[i]);
         }
+
+        if (U2STAbits.TRMT) { //if the transmit buffer is empty and register (no sending)
+            U2TXREG = popBQueue(&uart2_tx_queue); //trigger a send
+        }
     }
 }
 
 unsigned char readRXData(unsigned char interface)
 {
-    if (interface == 1 && uart1_status == INTERFACE_INITIALIZED) {
+    if (interface == 1 && (uart1_status & UART_RX_ENABLE)) {
         return popBQueue(&uart1_rx_queue);
-    } else if (interface == 2 && uart2_status == INTERFACE_INITIALIZED) {
+    } else if (interface == 2 && (uart2_status & UART_RX_ENABLE)) {
         return popBQueue(&uart2_rx_queue);
     }
     return -1;
 }
 
-#if INTERFACE_UART2_ENABLED
+unsigned int getTXSpace(unsigned char interface)
+{
+    if (interface == 1 && (uart1_status & UART_TX_ENABLE)) {
+        return getBQueueSpace(&uart1_tx_queue);
+    } else if (interface == 2 && (uart2_status & UART_TX_ENABLE)) {
+        return getBQueueSpace(&uart2_tx_queue);
+    }
+    return 0;
+}
 
 /**
  * Interrupt called whenever the TX buffer becomes empty
  */
-void __attribute__((__interrupt__, no_auto_psv)) _U2TXInterrupt(void)
-{
-    //while the TX buffer isn't full and we have data to send 
-    while (!U2STAbits.UTXBF && uart2_tx_queue.size != 0) {
-        U2TXREG = popBQueue(&uart2_tx_queue);
-    }
-    IFS1bits.U2TXIF = 0;
-}
+//void __attribute__((__interrupt__, no_auto_psv)) _U2TXInterrupt(void)
+//{
+//    //while the TX buffer isn't full and we have data to send 
+//    while (!U2STAbits.UTXBF && getBQueueSize(&uart2_tx_queue) != 0) {
+//        U2TXREG = popBQueue(&uart2_tx_queue);
+//    }
+//    IFS1bits.U2TXIF = 0;
+//}
 
 /**
  * Interrupt called when there is at least 1 character available to read from the rx buffer
  */
-void __attribute__((__interrupt__, no_auto_psv)) _U2RXInterrupt(void)
-{
-    //while the rx buffer has characters available to read
-    while (U2STAbits.URXDA) {
-        pushBQueue(&uart2_rx_queue, U2RXREG);
-    }
-}
-#endif
-
-#if INTERFACE_UART1_ENABLED
+//void __attribute__((__interrupt__, no_auto_psv)) _U2RXInterrupt(void)
+//{
+//    //while the rx buffer has characters available to read
+//    while (U2STAbits.URXDA) {
+//        pushBQueue(&uart2_rx_queue, U2RXREG);
+//    }
+//}
```

Comment:
because it'll interfere with the uart2 interrupts defined in net_inbound and net_outbound